lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4966,6 +4966,12 @@ namespace {
                     Impl.importSourceLoc(decl->getLocation()),
                     /*genericparams=*/nullptr, dc);
 
+      if (auto *GTD = dyn_cast<GenericTypeDecl>(typeDecl)) {
+        typealias->setGenericEnvironment(GTD->getGenericEnvironment());
+        if (GTD->isGeneric())
+          typealias->setGenericParams(GTD->getGenericParams()->clone(GTD));
```

Comment:
This is still wrong; it's supposed to be the *new* DeclContext, not the old one. That's `typealias`.

Reply:
I get it now, sorry - it has to use `typealias` because that's the scope for the new generic parameters, sorry about this...